src/app/student/map/editor/EditorMapComponent.tsx

```diff
@@ -0,0 +1,372 @@
+
+  const featureMap = useRef<FeatureMap>(
+    new Map(geoJsonData.features.map(feat => [feat.properties.id, feat]))
+  )
+
+  function resetFeatures(geoJsonData: GeoJsonBoothsData) {
+    featureMap.current = new Map(
+      geoJsonData.features.map(feat => [feat.properties.id, feat])
+    )
+    updateBooths()
+    updateDrawFeatures()
+  }
+
+  function setFeature(feat: GeoJsonBooth) {
+    featureMap.current.set(feat.properties.id, feat)
+  }
+  function deleteFeature(feat: GeoJsonBooth) {
+    featureMap.current.delete(feat.properties.id)
+  }
+
+  function onModeChange(e: { mode: DrawMode }) {
+    setDrawMode(e.mode)
+  }
+
+  function onSelectionChange(e: { features: GeoJsonBooth[] }) {
+    if (e.features.length === 0) return
+    if (activeFeature == null) setActiveFeatureId(e.features[0].properties.id)
+    else setActiveFeatureId(null)
+  }
+
+  function onCreate(e: { features: Feature<Polygon>[] }) {
+    e.features.forEach(feat => {
+      const id = Math.max(...Array.from(featureMap.current.keys())) + 1
+      const newFeat = {
+        ...feat,
+        id,
+        properties: { id, location: location.id, exhibitorId: -1 }
+      }
+      setFeature(newFeat)
+      setActiveFeatureId(id)
+      draw.delete(feat.id as string)
+      draw.add(newFeat)
+    })
+  }
+
+  function onUpdate(e: { features: GeoJsonBooth[] }) {
+    e.features.forEach(feat => setFeature(feat))
+    updateBooths()
+  }
+
+  function onDelete(e: { features: GeoJsonBooth[] }) {
+    e.features.forEach(feat => deleteFeature(feat))
+    updateBooths()
+  }
+
+  const draw = useMemo(() => {
+    return new MapboxDraw({
+      displayControlsDefault: false,
+      defaultMode: "simple_select"
+    })
+  }, [])
+
+  useEffect(() => {
+    if (!mapLoaded || !mapRef.current) return
+    const map = mapRef.current
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    if (!map.hasControl(draw)) map.addControl(draw, "top-left")
+
+    map.on("draw.create", onCreate)
+    map.on("draw.update", onUpdate)
+    map.on("draw.delete", onDelete)
+    map.on("draw.selectionchange", onSelectionChange)
+    map.on("draw.modechange", onModeChange)
+
+    return () => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      map.removeControl(draw)
+      map.off("draw.create", onCreate)
+      map.off("draw.update", onUpdate)
+      map.off("draw.delete", onDelete)
+      map.off("draw.selectionchange", onSelectionChange)
+      map.off("draw.modechange", onModeChange)
+    }
+  }, [mapLoaded, location])
+
+  function updateDrawFeatures() {
+    if (!mapLoaded) return
+    const currentFeatures: GeoJsonBoothsData = {
+      type: "FeatureCollection",
+      features: Array.from(featureMap.current.values()).filter(
+        feat => feat.properties.location === location.id
+      )
+    }
+    draw.set(currentFeatures)
+  }
+
+  useEffect(updateDrawFeatures, [location, mapLoaded])
```

Comment:
Nit: I general I think its good to have all useEffects in one place to have better overview